### ISSUE TITLE
`ConsumerMixin`: how to use the `channel` parameter in `get_consumers()` method?

### ISSUE THREAD

**Issue by pengye91** [celery/kombu#753]
Hi guys, I'm using `ConsumerMixin` to construct a general worker, which aims to contain several consumers using `get_consumers()` function. 

I found two problems when trying to implement this:

1. The `get_consumers()` method requires a channel argument, and I want the returned consumers on different channels, but it seems that I can't initialize the return consumer with a channel argument like this:
~~~ python
def get_consumers(self, Consumer, channel):
    return [
           Consumer(channel=channel_1, queues=[self.task_queue_1], callbacks=[self.on_task_1]),
           Consumer(channel=channel_2, queues=[self.task_queue_2], callbacks=[self.on_task_2]), 
    ]
~~~

And it will raise `TypeError: __init__() got multiple values for argument 'channel'`.

It seems the error should blame the `Consumer` contex manager in `ConsumerMixin` class:

~~~ python
    @contextmanager
    def Consumer(self):
        with self.establish_connection() as conn:
            self.on_connection_revived()
            info('Connected to %s', conn.as_uri())
            channel = conn.default_channel
            cls = partial(Consumer, channel,
                          on_decode_error=self.on_decode_error)
            with self._consume_from(*self.get_consumers(cls, channel)) as c:
                yield conn, channel, c
            debug('Consumers canceled')
            self.on_consume_end(conn, channel)
        debug('Connection closed')
~~~

It looks like no matter what channel you pass in, the context manager will always use the `deault_channel`.

So how can I create multiple consumers on different channels using `get_consumers` method? Or did I miss something?

2. As the first problem exists, I tried to return inialized consumers in `get_consumers()` method like this:
 
``` python
consumer_1 = Consumer(channel=channel_1, queues=[self.task_queue_1], callbacks=[self.on_task_1])
consumer_2 = Consumer(channel=channel_2, queues=[self.task_queue_2], callbacks=[self.on_task_2])

def get_consumers(self, Consumer, channel):
    return [consumer_1, consumer_2]
```

This code won't raise any error, but the consumers only start consuming when the whole running process breaks(press ctrl + C in my case). Why is that?

In conclusion, my main question is **how to create consumers on different channels using `get_consumers()` method.**

**Comment by pengye91** [comment]
After some struggling with the source code, I override the `Consumer` context manager in the `ConsumerMixin` class, now the channels in which consumers run can be controlled.
